torch/csrc/dynamo/guards.cpp

```diff
@@ -178,6 +187,45 @@ static PyObject* TensorGuards_new(
   return (PyObject*)self;
 }
 
+static std::vector<std::optional<int64_t>> c10IntArrayToVecOptInt(
```

Comment:
c10IntArray is a weird naming for it, since you could call this with std::vector<int64_t> (which would implicitly convert to int array ref). Try `wrapIntegersInOptional` suggested by GPT-4